### PR TITLE
services/horizon/ingest: OffersProcessor fuzzing function

### PR DESCRIPTION
**Pull request by bartekn**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Adds a fuzzing function to `OffersProcessor`. I used this one because ingestion code is actually in a processor (see https://github.com/stellar/go/issues/3196`).

### Known limitations

* A have a generator function for `corpus` but will push it to a separate repo (https://github.com/stellar/go/issues/3197).

### CODE REVIEW

**Comment by bartekn** [comment]
Generator source:
```go
package main

import (
	"bufio"
	"fmt"
	"os"

	"github.com/stellar/go/clients/horizonclient"
	"github.com/stellar/go/xdr"
)

func main() {
	client := horizonclient.DefaultPublicNetClient

	opRequest := horizonclient.OperationRequest{Order: horizonclient.OrderDesc, Limit: 200}
	ops, err := client.Operations(opRequest)
	if err != nil {
		panic(err)
	}

	records := ops.Embedded.Records
	txHashes := map[string]bool{}

	for _, op := range records {
		t := op.GetType()

		if t != "manage_sell_offer" && t != "manage_buy_offer" {
			continue
		}

		txHashes[op.GetTransactionHash()] = true
	}

	dir, err := os.Getwd()
	if err != nil {
		panic(err)
	}
	fmt.Println(dir)

	for hash := range txHashes {
		fmt.Println(hash)

		tx, err := client.TransactionDetail(hash)
		if err != nil {
			panic(err)
		}

		var meta xdr.TransactionMeta
		err = xdr.SafeUnmarshalBase64(tx.ResultMetaXdr, &meta)
		if err != nil {
			panic(err)
		}

		for i, opChanges := range meta.V2.Operations {
			f, err := os.Create("./corpus/" + fmt.Sprintf("%s-%d", hash, i))
			if err != nil {
				panic(err)
			}

			w := bufio.NewWriter(f)
			_, err = xdr.Marshal(w, opChanges.Changes)
			if err != nil {
				panic(err)
			}
			w.Flush()

			f.Close()
		}
	}
}

```

File: services/horizon/internal/ingest/processors/offers_processor_fuzz.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
+// +build gofuzz
```
Comment (2opremio) [inline services/horizon/internal/ingest/processors/offers_processor_fuzz.go:1]: sorry if this was discussed elsewhere, but wouldn't it be better to simply place the fuzzers in test files?

**Comment by 2opremio** [comment]
@bartekn do you still want to pursue this?

**Comment by bartekn** [comment]
@2opremio probably not as a part of this PR but a fuzzer for `OffersProcessor` should be part of the fuzzing suite @tamirms was working on.

**Comment by 2opremio** [comment]
Then, is it OK if we close the PR but keep the branch around?

**Comment by bartekn** [comment]
Yes, I think so.

**Pull request closed without merging**